src/Presentation/Controllers/IncomeController.cs

```diff
@@ -0,0 +1,100 @@
+using Microsoft.AspNetCore.Mvc;
+using System.Security.Claims;
+using ExpenseTrackerGrupo4.src.Domain.Entities;
+using ExpenseTrackerGrupo4.src.Infrastructure.Interfaces;
+
+
+namespace ExpenseTrackerGrupo4.src.Presentation.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class IncomeController : ControllerBase
+    {
+        private readonly IIncomeService _incomeService;
+
+        public IncomeController(IIncomeService incomeService)
+        {
+            _incomeService = incomeService;
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> CreateIncome([FromBody] Income income)
+        {
+            if (income == null)
+            {
+                return BadRequest("Income data is required.");
+            }
+
+            income.UserId = GetUserId();
+
+            await _incomeService.AddIncomeAsync(income);
+            return CreatedAtAction(nameof(GetIncomeById), new { id = income.Id }, income);
+        }
+
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<Income>>> GetIncomes()
+        {
+            var userId = GetUserId();
+            var incomes = await _incomeService.GetIncomeByIdAsync(userId);
+            return Ok(incomes);
+        }
+
+        [HttpGet("{id}")]
```

Comment:
Add [Authorize] tag to the endpoints because user should be authenticated.

Reply:
Solved